packages/core/src/hooks/useTransition.tsx

```diff
@@ -300,13 +306,35 @@ export function useTransition(
         }
         // Force update once idle and expired items exist.
         if (idle && transitions.some(t => t.expired)) {
+          /**
+           * Remove the exited transition from the list
+           */
+          exitingTransitions.current.delete(t)
```

Comment:
should this be in the `if (exitBeforeEnter)` check below? It doesn't seem to be affecting the functionality in the demo or anything, but as far as I can tell, `exitingTransitions` doesn't get populated unless `exitBeforeEnter === true` (I also may be missing something though 😅 ).

Reply:
Good point. But if someone changes the prop we might still have it I think... so maybe we should be checking if t exists. Although according to MDN it just returns `false` if it didn't exist.... 

So it won't break, but maybe it's a more explicit thing to do?

ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete

Reply:
Thats a good point about changing the prop -- since later in the hook `exitingTransition.scurrent.size` is used to check in the `useLayoutEffect`. I assume we'd want it to be empty no matter what then? And if it just returns `false` it won't affect anything else.

I think with that in mind it *is* kind of explicit -- "We want this removed when done, no matter what."